src/adrs/0129349218/README.md

```diff
@@ -0,0 +1,506 @@
+ *
+ * Messages MAY use {placeholders}. To prevent security vulnerabilities, these
+ * placeholders MUST only reference values from the `info.metadata` field.
+ */
+type MessageTemplate = string;
+
+/**
+ * Subject identifies the specific entity or element that caused the error
+ * within the context defined by `info.domain`. This provides a granular
+ * pointer to the root of the problem, whether it originates from user input,
+ * service-to-service communication, or an internal process.
+ *
+ * For example, when validating a message payload, the `subject` can be a
+ * JSON Pointer (`RFC 6901`) indicating the field that failed validation, such as
+ * `"/data/email"`. In other cases, it might be an application-specific
+ * identifier for a resource, like a user ID or transaction ID.
+ *
+ * This enables precise, programmatic error handling, allowing a client to
+ * highlight an invalid field in a UI or a backend service to trigger specific
+ * compensation logic.
+ */
+type Subject = string;
+
+/**
+ * ID identifies the error. The combination of info + id is unique for each
+ * distinct error event. Consumers MAY assume that error events with identical
+ * source and id are duplicates. This ID can be provided to support teams as a
+ * reference to help identify and track specific error occurrences, enabling
+ * faster troubleshooting and resolution of issues.
+ */
+type Id = string
+
+/**
+ * Time timestamp of when the occurrence happened.
+ */
+type time = Timestamp;
+
+/**
+ * Domain identifies the context in which an error happened.
+ */
+type ErrorInfo = {
+  /**
+   * The reason of the error. This is a constant value that identifies the
+   * proximate cause of the error. Error reasons are unique within a particular
+   * domain of errors. This should be at most 63 characters and match a regular
+   * expression of `[A-Z][A-Z0-9_]+[A-Z0-9]`, which represents UPPER_SNAKE_CASE.
+   */
+  reason: string;
+
+  /**
+   * The logical grouping to which the "reason" belongs. The error domain is
+   * typically the registered service name of the tool or product that generates
+   * the error. Example: "pubsub.googleapis.com". If the error is generated by
+   * some common infrastructure, the error domain must be a globally unique value
+   * that identifies the infrastructure. For Google API infrastructure, the error
+   * domain is "googleapis.com".
+   */
+  domain: string;
+
+  /**
+   * Additional structured details about this error. Values in this field MUST
+   * be carefully sanitized to avoid leaking sensitive information. See the
+   * "Secure Metadata Handling" section for detailed guidance.
+   */
+  metadata: Record<string, string>;
+};
+
+type HelpLink = {
+  /**
+   * Describes what the link offers.
+   */
+  description: string;
+
+  /**
+   * The URL of the link.
+   */
+  url: string;
+};
+
+type Help = {
+  /**
+   * URL(s) pointing to additional information on handling the current error.
+   */
+  links: Array<HelpLink>;
+};
```

Comment:
How do you envision these links being added to the object and maintained?

Reply:
Good question,

Personally, I would see this to be more a runtime specific concern while instancing the error than compile-time. Primarily of two things:

1. compile time would encode URLs that may change in the binary without be able to change it
2. some links required runtime context to make sense of the situation.

To make a point,

```elixir
docs_base_url = System.fetch_env!("DOCS_BASE_URL")
help = %{
  links: [
    %{url: external_url("/account/#{user_id}/status"), description: "..."},
    %{url: "#{docs_base_url}/errors/#{domain}/#{reason}"), description: "..."},
  ]
}
```

This depends on whatever `external_url` construction given the `user_id` does to have the final value, or knowing the final docs URL as well. So runtime required.